### DISCUSSION TITLE
Recommended way to provide direct guidance to Dosu?

### DISCUSSION THREAD

**Discussion by talmo** [dosu-ai/community#4] [Q&A]
Hi, we're just getting started with using Dosu and are in the stage of monitoring what it would reply to users before turning it loose on our repo.

In one of these, the reply was not quite right:

![CleanShot 2024-10-07 at 10 52 23@2x](https://github.com/user-attachments/assets/d83490e9-0c5b-4a26-9285-292b6b3f400d)

I want to provide some specific clarification -- namely that the `pipeline_form.yaml` file it's referring to isn't intended to be user-facing (it's a template we use for building GUIs). This is understandable and something that someone with less context about how the codebase is set up might look at for reference naively.

What's the best way to give the model highly specific feedback/instructions/corrections?

I know I can provide general instructions in the Workspace --> Description, but this seems to be a bit more general.

Should I set up a new data source and keep a running list?

If this were live, would replying to the bot post with specific corrections help or does it just listen to the thumbs up/down?

Thanks!

Talmo

**Comment by devstein** [comment]
Hey @talmo great question! There isn't a great way to provide direct guidance within the app yet. The best workaround is what you suggested (Workspace --> Description). 

> Should I set up a new data source and keep a running list?

What do you mean by this? 

> If this were live, would replying to the bot post with specific corrections help or does it just listen to the thumbs up/down?

Both are effective. We want to make these "corrections/learnings" available in the UI at some point, so you can directly edit them.
